front/app/containers/Admin/projects/project/projectHeader/ShareLink/index.tsx

```diff
@@ -0,0 +1,180 @@
+import React, { useState } from 'react';
+
+import {
+  Button,
+  Dropdown,
+  Box,
+  colors,
+  Text,
+  Icon,
+  IconButton,
+  Tooltip,
+} from '@citizenlab/cl2-component-library';
+
+import useAppConfiguration from 'api/app_configuration/useAppConfiguration';
+import useRefreshProjectPreviewToken from 'api/projects/useRefreshProjectPreviewToken';
+
+import useFeatureFlag from 'hooks/useFeatureFlag';
+
+import Divider from 'components/admin/Divider';
+
+import { trackEventByName } from 'utils/analytics';
+import { useIntl } from 'utils/cl-intl';
+
+import messages from '../messages';
+
+import RegenerateLinkModal from './RegenerateLinkModal';
+import tracks from './tracks';
+
+const ShareLink = ({
+  projectId,
+  projectSlug,
+  token,
+}: {
+  projectId: string;
+  projectSlug: string;
+  token: string;
+}) => {
+  const [refreshModalIsOpen, setRefreshModalIsOpen] = useState(false);
+  const isPreviewLinkEnabled = useFeatureFlag({ name: 'project_preview_link' });
+  const { data: appConfiguration } = useAppConfiguration();
+  const { mutate: refreshProjectPreviewToken, isLoading } =
+    useRefreshProjectPreviewToken();
+  const [shareDropdownIsOpen, setShareDropdownIsOpen] = useState(false);
+  const [linkIsCopied, setLinkIsCopied] = useState(false);
+
+  const link = `projects/${projectSlug}/preview/${token}`;
+
+  const handleCopyLink = () => {
+    navigator.clipboard.writeText(
+      `${appConfiguration?.data.attributes.host}/${link}`
+    );
+    setLinkIsCopied(true);
+    trackEventByName(tracks.copyProjectPreviewLink);
+  };
+
+  const { formatMessage } = useIntl();
+
+  const refreshLink = () => {
+    refreshProjectPreviewToken(
+      { projectId },
+      {
+        onSuccess: () => {
+          setLinkIsCopied(false);
+          setRefreshModalIsOpen(false);
+          trackEventByName(tracks.regenerateProjectPreviewLink);
+        },
+      }
+    );
+  };
+  return (
+    <Box position="relative">
+      <Button
+        id="e2e-share-link"
+        buttonStyle="secondary-outlined"
+        icon="share"
+        size="s"
+        padding="4px 8px"
+        onClick={() => {
+          setShareDropdownIsOpen(!shareDropdownIsOpen);
+          setLinkIsCopied(false);
+        }}
+      >
+        {formatMessage(messages.share)}
+      </Button>
+      <Dropdown
+        width="600px"
+        mobileWidth="300px"
+        zIndex="100000"
+        top="40px"
+        right="0px"
+        opened={shareDropdownIsOpen}
+        onClickOutside={() => {
+          setLinkIsCopied(false);
```

Comment:
Why don't we close the dropdown when clicking outside anymore?

Reply:
Because it interferes with the modal that opens when you recreate the preview link. Because clicking on the modal technically is outside of this dropdown, it gets closed and I don't like the UX in this case. So I made it now close when you click the button on top again, which I feel makes more sense.